### PR TITLE
[Torch] Fix deps issue for Torch

### PR DESCRIPTION
**Pull request by Wimmerer**
Attempt to fix https://github.com/JuliaRegistries/General/pull/68204 

Might also need to be a `BuildDependency` in addition to `Dependency`

### CODE REVIEW

**Comment by stemann** [comment]
When this Torch JLL is registered (https://github.com/JuliaRegistries/General/pull/68204), it might break current Torch.jl usages, as the C wrapper ("libtch") is not included - as there is no compat entry for Torch_jll in the currently registered Torch.jl.

File: T/Torch/build_tarballs.jl [hunk 1]
```diff
@@ -258,8 +258,7 @@ dependencies = [
     Dependency("SLEEF_jll", v"3.5.2"; compat = "3"),
 #    Dependency("TensorRT_jll"; platforms = cuda_platforms), # Building with TensorRT is not supported: https://github.com/pytorch/pytorch/issues/60228
     Dependency("XNNPACK_jll"; compat = "0.0.20210622"),
-    BuildDependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110"), v"3.13.0")),
-    HostBuildDependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110"), v"3.13.0")),
+    Dependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110"), v"3.13.0")),
```
Comment (giordano) [inline T/Torch/build_tarballs.jl:261]:
```suggestion
    Dependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110")), v"3.13.0"),
```
but I suspect the `HostBuildDependency` should still be there?

**Comment by giordano** [comment]
@gbaraldi I suspect https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/271 broke something:
```
ERROR: LoadError: TypeError: in keyword argument platforms, expected Vector{<:AbstractPlatform}, got a value of type Vector{Any}
Stacktrace:
 [1] top-level scope
   @ /agent/_work/1/s/T/Torch/build_tarballs.jl:247
in expression starting at /agent/_work/1/s/T/Torch/build_tarballs.jl:247
```

**Comment by gbaraldi** [comment]
I will take a look

File: T/Torch/build_tarballs.jl [hunk 2]
```diff
@@ -258,8 +258,7 @@ dependencies = [
     Dependency("SLEEF_jll", v"3.5.2"; compat = "3"),
 #    Dependency("TensorRT_jll"; platforms = cuda_platforms), # Building with TensorRT is not supported: https://github.com/pytorch/pytorch/issues/60228
     Dependency("XNNPACK_jll"; compat = "0.0.20210622"),
-    BuildDependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110"), v"3.13.0")),
-    HostBuildDependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110"), v"3.13.0")),
+    Dependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110")); compat="3.13.0"),
```
Comment (giordano) [inline T/Torch/build_tarballs.jl:261]: Not sure it's related to the current issue, but let's try to bring protoc back
```suggestion
    Dependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110")); compat="3.13.0"),
    HostBuildDependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110"), v"3.13.0")),
```

Comment (giordano) [reply T/Torch/build_tarballs.jl:261]: That was it indeed

**Comment by giordano** [comment]
> When this Torch JLL is registered (https://github.com/JuliaRegistries/General/pull/68204), it might break current Torch.jl usages, as the C wrapper ("libtch") is not included - as there is no compat entry for Torch_jll in the currently registered Torch.jl.

@stemann that should be fixed in the registry then (and in the package as well)

**Comment by giordano** [comment]
@Wimmerer maybe next time don't use "fix `<PR in the General registry>`", this closed https://github.com/JuliaRegistries/General/pull/68204 :wink:

**Comment by Wimmerer** [comment]
Wait what, I figured that would have privileges preventing it.

Yeah sorry about this PR in general, I think I made it right before sleeping then forgot 🥲

File: T/Torch/build_tarballs.jl [hunk 3]
```diff
@@ -258,7 +258,7 @@ dependencies = [
     Dependency("SLEEF_jll", v"3.5.2"; compat = "3"),
 #    Dependency("TensorRT_jll"; platforms = cuda_platforms), # Building with TensorRT is not supported: https://github.com/pytorch/pytorch/issues/60228
     Dependency("XNNPACK_jll"; compat = "0.0.20210622"),
-    BuildDependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110"), v"3.13.0")),
+    Dependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110")); compat="3.13.0"),
```
Comment (giordano) [inline T/Torch/build_tarballs.jl:261]: https://github.com/JuliaRegistries/General/pull/68946 sounds like protoc breaks the ABI across minor versions.  What a mess

**Pull request merged**